src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets

```diff
@@ -13,20 +13,37 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup>
 
     <!-- Analysis level is a single property that can be used to control both the compiler warning waves
-         and enable .NET analyzers  -->
+         and enable .NET analyzers. Valid values are 'none', 'latest', 'preview', or a version number  -->
+         
+    <!-- EffectiveAnalysisLevel is used to differentiate from user specified strings (such as 'none')
+         and an implied numerical option (such as '4')-->
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'none'">4.0</EffectiveAnalysisLevel>
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'latest'">5.0</EffectiveAnalysisLevel>
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'preview'">6.0</EffectiveAnalysisLevel>
+    
+    <!-- Automatically set AnalysisLevel if not specified -->
     <AnalysisLevel Condition="'$(AnalysisLevel)' == '' And
                               '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And
                               $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0'))">5.0</AnalysisLevel>
```

Comment:
Shouldn't we always default `AnalysisLevel` to `latest` instead of `5.0`?

Reply:
I think we can change this to `latest` and move it above the `EffectiveAnalysisLevel` group and never have to change this line again.